daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/RuntimeData.scala

```diff
@@ -672,7 +672,7 @@ sealed class ElementRuntimeData(
 
   def isSimpleType = optPrimType.isDefined
 
-  def schemaURIStringsForFullValidation = schemaURIStringsForFullValidation1.distinct
+  lazy val schemaURIStringsForFullValidation = schemaURIStringsForFullValidation1.distinct
   private def schemaURIStringsForFullValidation1: Seq[String] = (schemaFileLocation.uriString +:
     childERDs.flatMap { _.schemaURIStringsForFullValidation1 })
```

Comment:
Interesting algorithmic point about all this lazy val stuff we do.

If you change this recursion to call schemaURIStringsForFullValidation (without the "1" suffix), this algorithm will be greatly improved because we expect tons of duplicates - so squeezing them out with the "distinct" recursively, as we aggregate up the tree, that call should pay dividends.

Reply:
This is interesting indeed.  It wouldnt feel right burying that change in this PR, sounds like a great performance PR though.